[HttpFoundation] Fix erasing cookies issue

eiannone opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 4.2 (to be switched when merging)
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no     
| Deprecations? | no
| Tests pass?   | yes    
| Fixed tickets | #29675
| License       | MIT

Prevent replacing existing cookies when starting or regenerating session on PHP < 7.3 with 'cookie_samesite' option.
See issue #29675

eiannone committed:
Fix erasing cookies issue

Prevent replacing existing cookies when starting or regenerating session on PHP < 7.3 with 'cookie_samesite' option.
See issue https://github.com/symfony/symfony/issues/29675

nicolas-grekas commented:
@rpkamp ok for you?
Would it be possible to add a test case?

rpkamp commented:
LGTM

Although indeed an added test that proves the issue exists (and servers to prevent regressions) would be good :+1:

javiereguiluz committed:
Fixed minor typos in an error message

ogizanagi committed:
minor #29711 Fixed minor typos in an error message (javiereguiluz)

This PR was squashed before being merged into the 4.2 branch (closes #29711).

Discussion
----------

Fixed minor typos in an error message

| Q             | A
| ------------- | ---
| Branch?       | 4.2
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no     <!-- see https://symfony.com/bc -->
| Deprecations? | no <!-- don't forget to update UPGRADE-*.md and src/**/CHANGELOG.md files -->
| Tests pass?   | yes    <!-- please add some, will be required by reviewers -->
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Commits
-------

913fdbe107 Fixed minor typos in an error message

ro0NL committed:
[SecurityBundle] Fix traceable voters

Commit:
bug #29721 [SecurityBundle] Fix traceable voters (ro0NL)

This PR was squashed before being merged into the 4.2 branch (closes #29721).

Discussion
----------

[SecurityBundle] Fix traceable voters

| Q             | A
| ------------- | ---
| Branch?       | 4.2
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no     <!-- see https://symfony.com/bc -->
| Deprecations? |  no
| Tests pass?   | yes    <!-- please add some, will be required by reviewers -->
| Fixed tickets | #29385
| License       | MIT
| Doc PR        | symfony/symfony-docs#... <!-- required for new features -->

lets solve the BC break when autowiring individual voters.

Commits
-------

a7df429 [SecurityBundle] Fix traceable voters

xabbuh committed:
[VarExporter] fix exporting array indexes

fabpot committed:
bug #29741 [VarExporter] fix exporting array indexes (xabbuh)

This PR was merged into the 4.2 branch.

Discussion
----------

[VarExporter] fix exporting array indexes

| Q             | A
| ------------- | ---
| Branch?       | 4.2
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #29715
| License       | MIT
| Doc PR        |

Commits
-------

3c936f447b [VarExporter] fix exporting array indexes
